components/availability/src/Availability.svelte

```diff
@@ -1402,21 +1516,59 @@
   bind:this={main}
   bind:clientHeight
   class:ticked={show_ticks && view_as === "schedule"}
-  class:dated={allow_date_change}
   class:allow_booking
   on:mouseleave={() => endDrag(null, null)}
   style="--free-color: {free_color}; --busy-color: {busy_color}; --partial-color: {partial_color};"
 >
-  {#if allow_date_change}
-    <header class="change-dates">
-      <button on:click={goToPreviousDate} aria-label="Previous date"
-        ><BackIcon style="height:32px;width:32px;" /></button
-      >
-      <button on:click={goToNextDate} aria-label="Next date"
-        ><NextIcon style="height:32px;width:32px;" /></button
-      >
-    </header>
-  {/if}
+  <header class:dated={allow_date_change}>
+    <h2 class="month">
+      {#if days[days.length - 1].timestamp.getMonth() !== days[0].timestamp.getMonth()}
+        {#if days[days.length - 1].timestamp.getFullYear() !== days[0].timestamp.getFullYear()}
+          <!-- Case 1: months and years differ: Dec 2021 - Jan 2022 -->
+          {days[0].timestamp.toLocaleDateString("default", {
+            month: "short",
+            year: "numeric",
+          })}
+          -
+          {days[days.length - 1].timestamp.toLocaleDateString("default", {
+            month: "short",
+            year: "numeric",
+          })}
+        {:else}
+          <!-- Case 2: months differ, years the same: Oct - Nov 2021 -->
+          {days[0].timestamp.toLocaleDateString("default", {
+            month: "short",
+          })}
+          -
+          {days[days.length - 1].timestamp.toLocaleDateString("default", {
+            month: "short",
+            year: "numeric",
+          })}
+        {/if}
```

Comment:
This is such good level of detail 👍

Reply:
I love it! 

btw, When we start thinking of Components tech debt and readability I think this is the type of low hanging fruit we should look to abstract. I put this down in the Components Tech Debt story https://app.shortcut.com/nylas/story/70047/components-tech-debt